### PR TITLE
OCSP: add Request.validate and Response.validate to API

### PR DESCRIPTION
**Pull request by hannesm**
@NightBlues if you could take a brief look whether this looks good to you, I'd be happy. I'll get some unit tests for the validation part as well. Then it's good to be merged and released from my side.

### CODE REVIEW

**Comment by hannesm** [comment]
(EDIT: tests for OCSP.Response.validate are now in place)

**Comment by hannesm** [comment]
Similar to certificate, the raw bytes of an OCSP request are stored as well: it uses string encodings (that are not bijective in this library), that must be encoded the very same way for verifying the signature.

For the OCSP response this should likely be done similarly. The calls to `Certificate.Asn.certificate_to_cstruct` should be avoided (esp. if this is part of the to-be-signed data), since it leads to violation of the string encodings...

**Comment by hannesm** [comment]
all done now

**Comment by NightBlues** [comment]
looks good to me, but didn't tested :)

File: tests/ocsp.ml [hunk 1]
```diff
@@ -103,8 +106,11 @@ let test_simple_responder () =
     let responder_id = `ByName responder_dn in
     Fmt.epr "keytype = %a" Key_type.pp (Private_key.key_type responder_key);
```
Comment (torinnd) [inline tests/ocsp.ml:107]: This is the only case where there is output to stderr in this test module. Is that intentional or was this intended to be elided like line 70 above?

Comment (hannesm) [reply tests/ocsp.ml:107]: oops, I guess this is leftover debug code, and should be removed.

Comment (hannesm) [reply tests/ocsp.ml:107]: done in cd23240

**Pull request merged**